dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/IngestException.java

```diff
@@ -0,0 +1,48 @@
+package edu.harvard.iq.dataverse.ingest;
+
+import com.google.common.collect.Lists;
+import io.vavr.control.Option;
+
+import javax.ejb.ApplicationException;
+import java.util.ArrayList;
+import java.util.List;
+
+@ApplicationException(rollback = true)
+public class IngestException extends RuntimeException {
+
+    private String bundleKey;
+    private List<String> bundleArguments = new ArrayList<>();
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public IngestException(String message) {
+        super(message);
+    }
+
+    public IngestException(String message, String bundleKey) {
```

Comment:
Seems redundant to pass both the message and the key it was resolved from? To avoid typing problems (both message and the key are `String`s a factory method pattern can be utilized to resolve the bundle key instead of requiring the caller to do the resolution.

Reply:
We talked about it with Krzysiek, the idea is to have message as backup. If we ever do some kind of Bundle refactoring or anything happens to keys, this will be our backup message.

Reply:
I'm not sure what could happen to the keys other then them being changed? A separate bundle can be created with extra info where it's used and that in case of changes a database change set needs to also update database values. From code perspective the actual message in any language is irrelevant. What's relevant is the error classification that is done by the keys. I18n messages are only translations of those keys that can always be re-created - we have a backup of those in git already.